src/ivc/sangria/fold_relaxed_plonk_instance_chip.rs

```diff
@@ -186,12 +188,15 @@ impl<C: CurveAffine, const MARKERS: usize> AssignedRelaxedPlonkInstance<C, MARKE
             .try_into()
             .unwrap();
 
-        let folded_step_circuit_instances_hash_accumulator = gate.conditional_select(
-            region,
+        let folded_step_circuit_instances_hash_accumulator = match SCInstancesHashAcc::zip(
             lhs_step_circuit_instances_hash_accumulator,
             rhs_step_circuit_instances_hash_accumulator,
-            &condition,
-        )?;
+        ) {
+            SCInstancesHashAcc::Hash((lhs, rhs)) => {
+                SCInstancesHashAcc::Hash(gate.conditional_select(region, lhs, rhs, &condition)?)
```

Comment:
can you remind me the meaning of lhs, rhs and what does this line try to do?

Reply:
In Rust, `lhs` (left-hand side) and `rhs` (right-hand side) are common
parameter names used in functions involving binary operations or comparisons.

You can find a lot of usage of it inside Rust stdlib

Reply:
I know the lhs, rhs. My question is: mathematically, what lhs and rhs correspond to, or what does this function try to do

Reply:
It's `condition_select`, we just choose one of them here

It's condition select use at the end of the sfc logic
For zero step it will be the input acc
For other steps it will be result of fold fn